File: src/test/java/application/history/endpoint/history/api/GetAllHistoriesTest.java
package application.history.endpoint.history.api;

import application.history.IntegrationTest;
import application.history.WithFaker;
import application.history.history.application.services.GetAllHistoriesService;
import application.history.history.presentation.interfaces.api.GetAllHistoriesController;
import lombok.SneakyThrows;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.http.MediaType;
import org.springframework.test.web.servlet.ResultActions;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;


public class GetAllHistoriesTest extends IntegrationTest implements WithFaker {

    @Autowired
    private GetAllHistoriesService service;

    @SneakyThrows
    @Test
    public void getAllHistories() {
//        var randomContent = faker.random().hex();
//        var randomDate = faker.date().future(365, TimeUnit.DAYS).toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
//
//        CreateHistoryInput request = new CreateHistoryInput(randomContent, randomDate);
//
//        History history = new History(UUID.randomUUID().toString(), randomContent, randomDate);
//
//        given(service.run(new CreateHistoryInput(randomContent, randomDate))).willReturn(history);

//        .content(objectMapper.writeValueAsString(request))

        ResultActions resultActions = mvc.perform(
                get("/histories")
                .contentType(MediaType.APPLICATION_JSON)
                .accept(MediaType.APPLICATION_JSON)
        ).andExpect(status().isOk());

        var a = 1;
//
//        resultActions.andExpect(status().isCreated())
//                .andExpect(MockMvcResultMatchers.jsonPath("$.date").exists());
    }
}

File: src/test/kotlin/com/example/history/application/history/endpoint/history/api/GetAllHistoriesTest.kt
package com.example.history.application.history.endpoint.history.api

import application.history.IntegrationTest
import org.junit.jupiter.api.Test
import org.springframework.http.MediaType

class GetAllHistoriesTest : IntegrationTest(), WithFaker {
    @Autowired
    private val service: GetAllHistoriesService? = null
    //        var randomContent = faker.random().hex();
//        var randomDate = faker.date().future(365, TimeUnit.DAYS).toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
//
//        CreateHistoryInput request = new CreateHistoryInput(randomContent, randomDate);
//
//        History history = new History(UUID.randomUUID().toString(), randomContent, randomDate);
//
//        given(service.run(new CreateHistoryInput(randomContent, randomDate))).willReturn(history);

    //        .content(objectMapper.writeValueAsString(request))
    @get:Test
    @get:SneakyThrows
    val allHistories:
    //
//        resultActions.andExpect(status().isCreated())
//                .andExpect(MockMvcResultMatchers.jsonPath("$.date").exists());
            Unit
        get() {
//        var randomContent = faker.random().hex();
//        var randomDate = faker.date().future(365, TimeUnit.DAYS).toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
//
//        CreateHistoryInput request = new CreateHistoryInput(randomContent, randomDate);
//
//        History history = new History(UUID.randomUUID().toString(), randomContent, randomDate);
//
//        given(service.run(new CreateHistoryInput(randomContent, randomDate))).willReturn(history);

//        .content(objectMapper.writeValueAsString(request))
            val resultActions: ResultActions = mvc.perform(
                MockMvcRequestBuilders.get("/histories")
                    .contentType(MediaType.APPLICATION_JSON)
                    .accept(MediaType.APPLICATION_JSON)
            ).andExpect(MockMvcResultMatchers.status().isOk())
            val a = 1
            //
//        resultActions.andExpect(status().isCreated())
//                .andExpect(MockMvcResultMatchers.jsonPath("$.date").exists());
        }
}
